### PR TITLE
Added property onlyDeclaration to id-match rule (fixes #3488)

### LINKED ISSUE #3488: How to define id-match exception?
I am using the following pattern to enforce variable naming convention:

```
(^[A-Za-z]+(?:[A-Z][a-z]*)*\d*$)|(^[A-Z]+(_[A-Z]+)*(_\d$)*$)|(^(_|\$)$)
```

But it breaks in uncontrolled environment, e.g. testing frameworks. How to write a project-wide and file-wide list of exceptions to id-match?

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/26065929-how-to-define-id-match-exception?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by gajus**

### CODE REVIEW

File: docs/rules/id-match.md [hunk 1]
```diff
@@ -29,6 +29,11 @@ This rule needs a text RegExp to operate with, and accepts an options map. Its s
 1. `true` is the default and checks all property names
 2. `false` does not check property names at all (default)
 
+`onlyDeclaration` can have the following values:
+
```
Comment (nzakas) [inline docs/rules/id-match.md]: Probably better to be "onlyDeclarations` (plural) to match the style of other rules

File: tests/lib/rules/id-match.js [hunk 2]
```diff
@@ -21,6 +21,15 @@ var ruleTester = new RuleTester();
 ruleTester.run("id-match", rule, {
     valid: [
         {
+            code: "__foo = \"Matthieu\"",
+            options: [
```
Comment (nzakas) [inline tests/lib/rules/id-match.js]: Can you add an invalid test using this option as well?

**Comment by nzakas** [comment]
Overall looks good. Just a couple of small things to address.

File: docs/rules/id-match.md (hunk 1 above)
Comment (gajus) [inline docs/rules/id-match.md]: If this is a request to action something, then please phrase it that way. Now I cannot tell if you are thinking out loud, waiting for someone else to approve what you have said or actioning it.

File: tests/lib/rules/id-match.js (hunk 2 above)
Comment (gajus) [inline tests/lib/rules/id-match.js]: Will do.

**Comment by gajus** [comment]
@nzakas Travis build failed for what appears to be Travis-CI error. Whats the preferred way to restart the CI?

**Comment by gajus** [comment]
@nzakas Build is now passing.

**Comment by nzakas** [comment]
LGTM.

**Pull request merged**